crates/sui/src/ethereum_client_commands.rs

```diff
@@ -376,14 +378,15 @@ pub(crate) async fn eth_approve_message(
         None => return Err(anyhow!("ETH State object ID configuration not found")),
     };
```

Comment:
can you get this from `eth_dwallet_cap_id` - by querying the network?

Reply:
sure.